### ISSUE TITLE
Problem with JSONAPI 1.8 (TPS)

### ISSUE THREAD

**Issue by fabien91** [alecgorge/jsonapi#424]
Hello,

I have a problem with JSONAPI.
http://image.noelshack.com/fichiers/2015/05/1422557254-jsonapi.jpg

JSONAPI cause lag on my minecraft serveur.

**Comment by krisdb2009** [comment]
I agree. some methods like offline.user.names (something of that nature) seems to stop the main thread until the list is made. I wish it was ran in another thread so the entire server doesn't time out when its called...

**Comment by alecgorge** [comment]
The problem is that some methods *must* be called on the main thread because otherwise Minecraft throws a fit because it isn't threadsafe. There is currently no way to have some methods called on the main thread and others async. This needs to be fixed.
